src/longhorn_rpc_protocol.c

```diff
@@ -103,37 +111,57 @@ int receive_msg(int fd, struct Message *msg) {
                 errorf("fail to read magic version\n");
 		return -EINVAL;
         }
+
+	msg->MagicVersion = le16toh(msg->MagicVersion);
+
         if (msg->MagicVersion != MAGIC_VERSION) {
                 errorf("wrong magic version 0x%x, expected 0x%x\n",
                                 msg->MagicVersion, MAGIC_VERSION);
                 return -EINVAL;
         }
+
 	n = read_full(fd, &msg->Seq, sizeof(msg->Seq));
         if (n != sizeof(msg->Seq)) {
                 errorf("fail to read seq\n");
 		return -EINVAL;
         }
+
+	msg->Seq = le32toh(msg->Seq);
+
         n = read_full(fd, &msg->Type, sizeof(msg->Type));
         if (n != sizeof(msg->Type)) {
                 errorf("fail to read type\n");
 		return -EINVAL;
         }
-        n = read_full(fd, &msg->Offset, sizeof(msg->Offset));
-        if (n != sizeof(msg->Offset)) {
+
+	msg->Type = le32toh(msg->Type);
+
+        n = read_full(fd, &Offset, sizeof(Offset));
+        if (n != sizeof(Offset)) {
```

Comment:
@keithalucas 
Should it be `msg->Offset` rather than `Offset`?

Reply:
I read the value into a temporary unsigned variable then convert it to the signed value a little after this.